query-engine/js-connectors/src/error.rs

```diff
@@ -0,0 +1,47 @@
+use futures::{Future, FutureExt};
+use napi::Error as NapiError;
+use quaint::error::Error as QuaintError;
+use std::{any::Any, panic::AssertUnwindSafe};
+
+/// transforms a napi error into a quaint error copying the status and reason
+/// properties over
+pub(crate) fn into_quaint_error(napi_err: NapiError) -> QuaintError {
+    let status = napi_err.status.as_ref().to_owned();
+    let reason = napi_err.reason.clone();
+
+    QuaintError::raw_connector_error(status, reason)
+}
+
+/// catches a panic thrown during the executuin of an asynchronous closure and transforms it into
+/// the Error variant of a napi::Result.
+pub(crate) async fn async_unwinding_panic<F, R>(fut: F) -> napi::Result<R>
+where
+    F: Future<Output = napi::Result<R>>,
+{
+    match AssertUnwindSafe(fut).catch_unwind().await {
+        Ok(result) => result,
+        Err(panic_payload) => panic_to_napi_err(panic_payload),
+    }
```

Comment:
It works, but the `match` could be simplified with an `.unwrap_or_else(panic_to_napi_err)`. The rest of the codebase seems to lean into `match` only when the alternative would be too convoluted